libmscore/chordrest.cpp

```diff
@@ -1218,6 +1218,38 @@ QString ChordRest::accessibleExtraInfo() const
       return rez;
       }
 
+//---------------------------------------------------------
+//   isMelismaEnd
+//    returns true if chordrest represents the end of a melisma
+//    TODO: make this member function of ChordRest
+//    and if possible, calculate and cache the value during lyrics layout
+//    then merely retrieve it here
+//---------------------------------------------------------
+
+bool isMelismaEnd(const ChordRest* cr)
+      {
+      Fraction t = cr->tick();
+      int tr = cr->track();
+      if (!cr->lyrics().empty())
+            return false;
+      while (true) {
+            Segment* s = cr->segment()->prev1();
+            if (!s)
+                  return false;
+            cr = s->nextChordRest(tr, true);
+            if (!cr)
+                  return false;
+            if (!cr->lyrics().empty()) {
+                  Lyrics* l = cr->lyrics().front();
+                  if (l->endTick() == t)
+                        return true;
+                  else
+                        return false;
+                  }
+            }
+      return false;
```

Comment:
Isn't this line unreachable? The only way the above while loop ends is by returning.

Reply:
Yes :-).  This code is just temporary, I hope to replace it with code that doesn’t need to loop in this way at all, using the approach mentioned in the TODO

Reply:
Yes :-).  This code is just temporary, I hope to replace it with code that doesn’t need to loop in this way at all, using the approach mentioned in the TODO